shopinvader_search_engine_update_image/models/fs_product_image.py

```diff
@@ -8,5 +8,20 @@ class FsProductImage(models.Model):
     _name = "fs.product.image"
     _inherit = ["fs.product.image", "se.product.update.mixin"]
 
+    def _update_se_thumbnails(self):
+        for rec in self:
+            for thumbnail in rec.image.attachment.se_thumbnail_ids:
+                values = thumbnail._prepare_tumbnail(
+                    rec.image, thumbnail.size_x, thumbnail.size_y, thumbnail.base_name
+                )
+                thumbnail.write(values)
+
+    def write(self, vals):
+        needs_update = self.needs_product_update(vals)
+        res = super().write(vals)
+        if needs_update and "specific_image" in vals:
+            self._update_se_thumbnails()
+        return res
```

Comment:
Shouldn't we add this in the relation mixin? Because we will have to add thoses inside all submodels of the mixin